nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexMojo.java

```diff
@@ -80,28 +82,24 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
         // If we add clean exceptions to allExceptions then the build fails if anything fails without nondex.
         // Everything in nondex-test is expected to fail without nondex so we throw away the result here.
-        for (int j = 0; j < this.numRunsWithoutShuffling; j++) {
-            CleanSurefireExecution cleanExec = new CleanSurefireExecution(
-                this.surefire, this.originalArgLine, this.mavenProject,
-                    this.mavenSession, this.pluginManager,
-                    Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_DIR).toString());
-            this.executeSurefireExecution(allExceptions, cleanExec);
-            this.executionsWithoutShuffling.add(cleanExec);
-        }
 
-        for (int i = 0; i < this.numRuns; i++) {
-            NonDexSurefireExecution execution =
-                    new NonDexSurefireExecution(this.mode, this.computeIthSeed(i),
-                            Pattern.compile(this.filter), this.start, this.end,
-                            Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_DIR).toString(),
-                            Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_JAR_DIR)
-                                .toString(),
-                            this.surefire, this.originalArgLine, this.mavenProject,
-                            this.mavenSession, this.pluginManager);
-            setFilePath(execution.getConfiguration().getRunFilePath());
-            this.executions.add(execution);
-            allExceptions = this.executeSurefireExecution(allExceptions, execution);
-            this.writeCurrentRunInfo(execution);
+        if (this.surefire.getExecutions() != null && !this.surefire.getExecutions().isEmpty()) {
+            // We need to do this because Xpp3Dom doesn't accept null yet setConfiguration accepts null.
+            Xpp3Dom origNode = null;
+            if (this.surefire.getConfiguration() != null) {
+                origNode = (Xpp3Dom) this.surefire.getConfiguration();
+            }
+            for (PluginExecution exec: this.surefire.getExecutions()) {
+                if (exec.getConfiguration() != null) {
+                    this.surefire.setConfiguration((Xpp3Dom) exec.getConfiguration());
+                } else {
+                    this.surefire.setConfiguration(origNode);
+                }
+                allExceptions = this.runExecutions(allExceptions);
```

Comment:
I understand you copied this logic from before, but was it ever good? The name `allExceptions` implies as if it collects exceptions across all runs (whether multiple runs of one Surefire config with many random seeds, or multiple runs due to multiple Surefire configs). However, the logic seems to be getting just the last exception. Does `executeSurefireExecution` somehow appends to the result?